python/lsst/meas/deblender/plugins.py

```diff
@@ -115,6 +116,205 @@ def __str__(self):
+        getattr(pkResult, flag)()
+
+def build_multiband_templates(debResult, log, useWeights=False, usePsf=False,
+                              useStrictMonotonicity=True, truncate=True, **multiband_kwargs):
+    """Run the Multiband Deblender to build templates
+
+    Parameters
+    ----------
+    debResult: `lsst.meas.deblender.baseline.DeblenderResult`
+        Container for the final deblender results.
+    log: `log.Log`
+        LSST logger for logging purposes.
+    useWeights: bool, default=False
+        Whether or not to use the variance map in each filter for the fit.
+        This is currently not implemented
+    usePsf: bool, default=False
+        Whether or not to convolve the image with the PSF in each band.
+        This is not yet implemented in an optimized algorithm, so it is recommended
+        to leave this term off for now
+    useStrictMonotonicity: bool, default=True
+        Whether or not to use the strict monotonicity proximal operator (see Melchior et al. 2017)
+    truncate: bool, default=True
+        Whether to truncate the image (default) or add an extra row/column if the dimensions of the
+        `Footprint` bounding box are not odd.
+    multiband_kwargs: dict
+        Additional parameters to pass to the deblend package.
+
+    Returns
+    -------
+    modified: `bool`
+        If any templates have been created then ``modified`` is ``True``,
+        otherwise it is ``False`` (meaning all of the peaks were skipped).
+    """
+    import deblender
+
+    # Extract coordinates from each MultiColorPeak
+    bbox = debResult.footprint.getBBox()
+    peakSchema = debResult.footprint.peaks.getSchema()
+    xmin = bbox.getMinX()
+    ymin = bbox.getMinY()
+    peaks = [[pk.x-xmin, pk.y-ymin] for pk in debResult.peaks]
+
+    # Create the data array from the masked images
+    maskedImages = [mimg.Factory(mimg, debResult.footprint.getBBox(), PARENT)
+                    for mimg in debResult.maskedImages]
+    data = np.array([mimg.image.array for mimg in maskedImages])
+    data = deblender.nmf.reshape_img(data, truncate=truncate)
+
+    default_kwargs = {
+        "components": None,
+        "constraints": "S",
+    }
+    
+    for k,v in default_kwargs.items():
+        if k not in multiband_kwargs.keys():
+            multiband_kwargs[k] = v
+
+    # Update optional parameters
+    if "weights" not in multiband_kwargs:
+        if useWeights:
+            # Use the inverser variance as the weights
+            weights = 1/np.array([mimg.variance.array for mimg in maskedImages])
+
+            # Set bad pixeks weights to zero
+            maskPlane = maskedImages[0].getMask().getMaskPlaneDict()
+            badPixels = (1<<maskPlane["BAD"] |
+                         1<<maskPlane["CR"] |
+                         1<<maskPlane["NO_DATA"] |
+                         1<<maskPlane["SAT"] |
+                         1<<maskPlane["SUSPECT"])
+            fpMask = afwImage.Mask(bbox)
```

Comment:
I think that there has to be an inversion either way, since `debResult.footprint.spans.setMask(fpMask, 1)` requires the mask to be inverted from the weights.